### DISCUSSION TITLE
export security events to CSV

### DISCUSSION THREAD

**Discussion by iptwins** [wazuh/wazuh#17570] [Q&A]
Hello community, 

I used to have an old version of Wazuh and I could (once I had selected an agent and a date range) export the security events and groups to CSV.
With the latest version, I don't have this option, the only export I have is a PDF which only corresponds to the screen display.
I've looked a bit in the manual, but I can't find how to do it.
Thank you for your help.

**Comment by mdiego92** [comment]
Hello @iptwins,

To export data to CSV, please follow these steps:

1) Go to Discover.
![image](https://github.com/wazuh/wazuh/assets/62477966/406241a7-d673-490f-b482-a857818c9ce4)
2) Set up the required time and field filters.
3) Click on Save in the top right corner and save the search.
![image](https://github.com/wazuh/wazuh/assets/62477966/72d8a1c0-9e7a-488a-980f-3c20c8fbdd90)
4) Click on Reporting in the top right corner and Generate CSV.
![image](https://github.com/wazuh/wazuh/assets/62477966/cde97589-3f72-4af6-b870-ccef5b37062e)

Hopefully, this works for you!
Please let me know if you have any remaining questions.

**Comment by iptwins** [comment]
Thank you very much !
so I did a test, I'm in "Discover", I've created my search and saved it, as well as a timeline.
But when I click on reporting, the Genetate CSV and Create report definition links are greyed out ... I tried modifying the timeline (too many results maybe) but it doesn't change anything.
Thank you for your help.

![Capture d'écran 2023-06-13 174649](https://github.com/wazuh/wazuh/assets/55782703/4b1d7e20-7887-498e-8cf0-624c5c27bb0c)

**Reply by mdiego92** [reply]
Hello @iptwins,

The last action prior to generating the CSV report should be saving the search.
Regarding the number of hits, there is a 10,000 CSV row limit.

Could you please confirm if you're able to do it after setting a lower number of results?

Looking forward to your response!

**Comment by iptwins** [comment]
Thank you for your reply. I've redone the test but it doesn't work.
I took a default period of the day.
I did a search with the name of my machine, then once the results were displayed, I saved the search.
Then I clicked on reporting, and this is what I got:
![image](https://github.com/wazuh/wazuh/assets/55782703/6496f653-7ac5-4d7f-ad6d-768db8ed7e88)
although i don't have too much data (1700) , i don't understand.

**Comment by mdiego92** [comment]
Hello @iptwins,

Could you please try doing it from the Reporting section?

1) Go to Reporting.
![image](https://github.com/wazuh/wazuh/assets/62477966/9c7fcb07-5ce4-47aa-84d9-e825e4bc89ba)
2) Click on Create in the Report Definition Area.
3) Complete with a name and select your Saved Search after changing the setting from the Report Source.
![image](https://github.com/wazuh/wazuh/assets/62477966/e97fe80d-d607-410c-aef6-147ef968a75b)

Looking forward to your feedback!
Regards.

**Comment by iptwins** [comment]
thank you for taking the time to reply and try to help me.
I've done what you asked, but there's no search in the drop-down menu, so I must be missing something ...
here, I do have a saved search (I didn't take any action afterwards):
![Capture d'écran 2023-06-14 213222](https://github.com/wazuh/wazuh/assets/55782703/4d8778b0-07fe-4604-a337-97202d1d2bd3)
here the attempt to create the report which doesn't offer me a list of searches:
![Capture d'écran 2023-06-14 213251](https://github.com/wazuh/wazuh/assets/55782703/b05abb32-fd73-4b9d-be9c-9981e17cbd80)

thank you.
Regards.

**Reply by mdiego92** [reply]
Hello @iptwins,

It seems the issue is how you save your results! Both methods should work if you use the Save button on the top right corner:
![image](https://github.com/wazuh/wazuh/assets/62477966/ed05e52d-1317-4dff-aec2-3df79dcfe014)

Please let me know if that worked for you!
Regards!

**Comment by iptwins** [comment]
Hello, 
I've just realised, I was using the (diskette) button at the top left...  
I used the "save" button that you indicate, everything is ok!
Thank you very much for your help and patience. 
👍

**Reply by mdiego92** [reply]
Hello @iptwins,

I'm glad the issue is now resolved!
